File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if not s:
            return s
        new_s = "#"
        for ch in s:
            new_s = new_s + ch + "#"
        curr = new_s[0]
        for i in range(len(new_s)):
            l = i - 1
            r = i + 1
            while(l >= 0 and r < len(new_s)):
                if new_s[l] == new_s[r]:
                    if (r - l + 1) > len(curr):
                        curr = new_s[l:r+1]
                    l -= 1 
                    r += 1
                else:
                    break
        res = ""
        for ch in curr:
            if ch != "#":
                res+=ch
        return res

File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        int res = 0;
        int maxl = 0;
        int maxr = 0;
        int count = 0;
        for(int i = 0;i<s.length();i++){
            int l = i;
            int r = i;
            count = 0;
            while(l>=0 && r < s.length()){
                if(s.charAt(l) == s.charAt(r)){
                    count+=1;
                }
                else{
                    break;
                }
                if(count > res){
                    res = count;
                    maxl = l;
                    maxr = r;
                }
                l-=1;
                r+=1;
            }
            
        }
        for(int i = 0;i<s.length();i++){
            int l = i;
            int r = i+1;
            count = 1;
            while(l>=0 && r < s.length()){
                if(s.charAt(l) == s.charAt(r)){
                    count+=1;
                }
                else{
                    break;
                }
                if(count > res){
                    res = count;
                    maxl = l;
                    maxr = r;
                }
                l-=1;
                r+=1;
            }
        }
        return s.substring(maxl,maxr+1);
    }
}